scripts/check-update.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+set -eio pipefail
+
+usage() {
+    echo "Usage:"
+    echo "  $0 AMI_TYPE"
+    echo "Example:"
+    echo "  $0 al2"
+    echo "AMI_TYPE Must be one of: al1, al2, al2023"
+}
+
+error() {
+    local msg="$1"
+    echo "ERROR: $msg"
+    usage
+    exit 1
+}
+
+readonly ami_type="$1"
+if [ -z "$ami_type" ]; then
+    error "AMI_TYPE must be provided"
+fi
+
+case "$ami_type" in
+"al1")
+    cp release-al1.auto.pkrvars.hcl release-al1.old.hcl
+    ./generate-release-vars.sh al1
+    diff_val=$(diff <(grep -v ami_version release-al1.old.hcl) <(grep -v ami_version release-al1.auto.pkrvars.hcl))
+    if [ -z "$diff_val" ]; then
+        UpdateAl1="false"
+    else
+        UpdateAl1="true"
+    fi
+
+    if [ "$UpdateAl1" != "true" ]; then
+        UpdateAl1=$(./check-update-security.sh al1)
+    fi
+    if [ "$UpdateAl1" = "true" ]; then
+        git add release-al1.auto.pkrvars.hcl
+        echo "Al1 Has Updates"
+    fi
+    ;;
+"al2")
+    cp release-al2.auto.pkrvars.hcl release-al2.old.hcl
+    ./generate-release-vars.sh al2
+    diff_val=$(diff <(grep -v ami_version release-al2.old.hcl) <(grep -v ami_version release-al2.auto.pkrvars.hcl))
+    if [ -z "$diff_val" ]; then
+        UpdateAl2="false"
+    else
+        UpdateAl2="true"
+    fi
+
+    if [ "$UpdateAl2" != "true" ]; then
+        UpdateAl2=$(./check-update-security.sh al2)
+        if [ "$UpdateAl2" != "true" ]; then
+            UpdateAl2=$(./check-update-security.sh al2_arm)
+        fi
+    fi
+    if [ "$UpdateAl2" = "true" ]; then
+        git add release-al2.auto.pkrvars.hcl
+        echo "Al2 Has Updates"
+    fi
```

Comment:
Do you think we can capture these actions in a function and call it three times for AL1, AL2, and AL2023?

Reply:
I modified it so its much cleaner now, thanks for pointing this out. Instead of a case by case basis, the file now runs the whole thing for the specific AMI with the same code.